lib/forwardanalyzer.cpp

```diff
@@ -25,12 +25,9 @@ struct ForwardTraversal {
 
     std::pair<bool, bool> evalCond(const Token* tok) {
         std::vector<int> result = analyzer->evaluate(tok);
-        bool checkThen = std::any_of(result.begin(), result.end(), [](int x) {
-            return x;
-        });
-        bool checkElse = std::any_of(result.begin(), result.end(), [](int x) {
-            return !x;
-        });
+        // TODO: We should convert to bool
+        bool checkThen = std::any_of(result.begin(), result.end(), [](int x) { return x == 1; });
```

Comment:
It should be `x != 0` right?

Reply:
It should be, but this test in valueflow fails:

```cpp
void f(int a) { // #6826
    int x = a ? a : 87;
    if (a && x) {} // <- x should not be 87
}
```

Hence the TODO.

Reply:
The reason this test started to fail is that the [`evaluate`](https://github.com/danmar/cppcheck/pull/2878/files#diff-16a9ec3ff1314646e106eb413234f5614e3a7f527fb82d0a369755bee65a7e31R2250) function was changed to only call `conditionIsTrue` or `conditionIsFalse` on tokens with `&&` or `||` otherwise it calls `execute` directly. 

The `conditionIsTrue` function only checks the result is `1` so it does the same here to maintain the same behavior.